### PR TITLE
chore: migrate ViewHeader components to css modules

### PR DESCRIPTION
**Pull request by uladzimirdev**
[epic](https://github.com/metabase/metabase/issues/50983)

### Description

PR 4/?

Rework ViewHeader components to css modules

### How to verify

Check view header in QB

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.module.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.HeaderBreadcrumbs {
+  color: var(--mb-color-text-light);
+  font-size: 0.8em;
+  font-weight: bold;
+  padding-left: 0.5em;
+  padding-right: 0.5em;
+  user-select: none;
+}
+
+.HeaderBadge :global(.Icon) {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.module.css:10]: Can we avoid this `:global`?

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx [hunk 2]
```diff
@@ -40,22 +53,35 @@ export function HeadBreadcrumbs({
...
+function Divider({ char = "/" }) {
+  return (
+    <Box component="span" className={HeaderBreadcrumbsS.HeaderBreadcrumbs}>
+      {char}
+    </Box>
   );
 }
 
 HeadBreadcrumbs.Badge = HeaderBadge;
+// TODO: likely not used and can be removed
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx:86]: Yup, let's remove it :+1:

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/SavedQuestionLeftSide/SavedQuestionLeftSide.1.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import cx from "classnames";
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/SavedQuestionLeftSide/SavedQuestionLeftSide.1.tsx:1]: Is this `SavedQuestionLeftSide.1.tsx` file a leftover?

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx [hunk 4]
```diff
@@ -40,22 +53,35 @@ export function HeadBreadcrumbs({
             {isValidElement(part) ? (
               part
             ) : (
-              <HeaderBadge
+              <Badge
+                className={cx(HeaderBreadcrumbsS.HeaderBadge, {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx:57]: :x: Font size is off in collection name:

image

:x: and also in question title when there are filters

image

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx [hunk 5]
```diff
@@ -40,22 +53,35 @@ export function HeadBreadcrumbs({
...
+                  [HeaderBreadcrumbsS.headVariant]: variant === "head",
+                })}
                 to={part.href}
                 icon={part.icon}
                 inactiveColor={badgeInactiveColor}
               >
                 {part.name}
-              </HeaderBadge>
+              </Badge>
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx:65]: :x: The description should be visible when a question is opened, it's only visible after hover.
On mouseleave the description should disappear and the title should get vertically centered.

https://github.com/user-attachments/assets/e0a7a8ac-0290-46e3-95f7-059d3ed69a3e

**Review by kamilmielnik — CHANGES_REQUESTED** [review]
Looks good but I found some small visual regressions

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx (hunk 2 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx]: removed

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/SavedQuestionLeftSide/SavedQuestionLeftSide.1.tsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/SavedQuestionLeftSide/SavedQuestionLeftSide.1.tsx]: removed

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx]: Both should be fixed

image

image

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx (hunk 5 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/HeaderBreadcrumbs/HeaderBreadcrumbs.jsx]: should be fixed 
https://github.com/user-attachments/assets/f23bbd88-0241-41a3-b71a-8fd84072a58b

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**